src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/ChainPal.cs

```diff
@@ -51,6 +53,12 @@ internal SecTrustChainPal()
                 out chain,
                 out osStatus);
 
+            if (trustMode == X509ChainTrustMode.CustomRootTrust)
+            {
+                SafeCreateHandle customCertsArray = null;
+                customCertsArray = PrepareCustomCertsArray(customTrustStore);
+                Interop.AppleCrypto.X509ChainSetTrustAnchorCertificates(chain, customCertsArray);
```

Comment:
Who owns the SafeCreateHandle at this point?  Was ownership transferred?  Should we be disposing of it?